From 9f2e4a1027ee90d9fec6f3144615586ab5f399de Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Tue, 6 Jun 2017 18:44:16 +0800
Subject: [PATCH 1881/5242] MLK-15031-07 tty: serial: fsl_lpuart: add
 port.icount for DMA rx path

commit  398a779e78911ed64b8e26e430495174059bf62a from
https://source.codeaurora.org/external/imx/linux-imx.git

Add port.icount to stat. the DMA RX received count.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Reviewed-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/fsl_lpuart.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/tty/serial/fsl_lpuart.c b/drivers/tty/serial/fsl_lpuart.c
index c022fb5..b14ea4e 100644
--- a/drivers/tty/serial/fsl_lpuart.c
+++ b/drivers/tty/serial/fsl_lpuart.c
@@ -981,10 +981,9 @@ static void lpuart_copy_rx_to_tty(struct lpuart_port *sport,
 	copied = tty_insert_flip_string(tty,
 			((unsigned char *)(sport->dma_rx_buf_virt)), count);
 
-	if (copied != count) {
-		WARN_ON(1);
-		dev_err(sport->port.dev, "RxData copy to tty layer failed\n");
-	}
+	if (copied != count)
+		sport->port.icount.buf_overrun += count - copied;
+	sport->port.icount.rx += copied;
 }
 
 static void lpuart_dma_stop(struct lpuart_port *sport)
-- 
1.7.9.5

